packages/jellyfish-api-core/__tests__/category/loan/tokenSplits.test.ts

```diff
@@ -0,0 +1,966 @@
+  }
+  if (!poolpairNew.reserveB.eq(ppInfoBefore.reserveB)) {
+    expect(poolpairNew.reserveB).toStrictEqual(ppInfoBefore.reserveB.multipliedBy(2))
+  }
+
+  expect(poolpairNew.tradeEnabled).toStrictEqual(true)
+}
+
+async function setupPools (): Promise<void> {
+  // create TSLA-DFI
+  await testing.poolpair.create({
+    tokenA: 'TSLA',
+    tokenB: 'DFI'
+  })
+  await testing.generate(1)
+
+  // add TSLA-DFI
+  await testing.poolpair.add({
+    a: { symbol: 'TSLA', amount: 100 },
+    b: { symbol: 'DFI', amount: 200 }
+  })
+  await testing.generate(1)
+
+  const ppTokenID = Object.keys(await testing.rpc.token.getToken('TSLA-DFI'))[0]
+
+  // setup govvars
+  await testing.rpc.masternode.setGov({
+    ATTRIBUTES: {
+      [`v0/poolpairs/${ppTokenID}/token_a_fee_pct`]: '0.01',
+      [`v0/poolpairs/${ppTokenID}/token_b_fee_pct`]: '0.03'
+    }
+  })
+
+  await testing.rpc.masternode.setGov({ LP_SPLITS: { [Number(ppTokenID)]: 1 } })
+  await testing.rpc.masternode.setGov({ LP_LOAN_TOKEN_SPLITS: { [Number(ppTokenID)]: 1 } })
+  await testing.generate(1)
+}
+
+describe('Token splits', () => {
+  beforeEach(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+    await setup()
+  })
+
+  afterEach(async () => {
+    await testing.container.stop()
+  })
+
+  it('should check token after split', async () => {
+    const tokenTSLA = await testing.rpc.token.getToken(tslaID)
+    const newMintedTSLA = tokenTSLA[tslaID].minted.multipliedBy(2)
+
+    // set a token split at current block + 2 for TSLA and FB
+    const splitBlock = await testing.rpc.blockchain.getBlockCount() + 2
+    await testing.rpc.masternode.setGov({
+      ATTRIBUTES: {
+        [`v0/oracles/splits/${splitBlock}`]: `${tslaID}/2`
+      }
+    })
+    await testing.generate(2)
+    await checkTokenSplit(tslaID, 'TSLA', '/v1', newMintedTSLA, true, false)
+  })
+
+  it('should check active vault after split', async () => {
+    const tokenFB = await testing.rpc.token.getToken(fbID)
+    const newMintedFB = tokenFB[fbID].minted.multipliedBy(2)
+
+    const vaultFBBefore = await testing.rpc.vault.getVault(vaultId4) as VaultActive
+
+    // set a token split at current block + 2 for FB
+    const splitBlock = await testing.rpc.blockchain.getBlockCount() + 2
+    await testing.rpc.masternode.setGov({
+      ATTRIBUTES: {
+        [`v0/oracles/splits/${splitBlock}`]: `${fbID}/2`
+      }
+    })
+    await testing.generate(2)
+    const newFBID = await checkTokenSplit(fbID, 'FB', '/v1', newMintedFB, true, false)
+
+    {
+      // check vaults
+      const vaultFBAfter = await testing.rpc.vault.getVault(vaultId4) as VaultActive
+      expect(vaultFBAfter.vaultId).toStrictEqual(vaultFBBefore.vaultId)
+      expect(vaultFBAfter.ownerAddress).toStrictEqual(vaultFBBefore.ownerAddress)
+      expect(vaultFBAfter.state).toStrictEqual(VaultState.FROZEN)
```

Comment:
`FROZEN` because of *lock* ??

Reply:
yes i believe. But curious why vault state in `IN_LIQUIDATION` after unlock. This is not how initial test was written as i can remember🤔

Reply:
https://github.com/JellyfishSDK/jellyfish/commit/40963750da49bd24fbeb5782648ad1e546870935